backend/src/auth/handlers.rs

```diff
@@ -35,7 +35,19 @@ pub(crate) async fn handle_login(req: Request<Body>, ctx: &Context) -> Result<Re
 
             Response::builder()
                 .status(StatusCode::NO_CONTENT)
-                .header("set-cookie", session_id.set_cookie().to_string())
+                .header("set-cookie", session_id.set_cookie(
+                    // The `cookie` crate unfortunately uses `time::Duration`
+                    // which can represent negative durations for the price
+                    // of being able to represent very long durations.
+                    // Thus, conversions from `std::time::Duration` can fail.
+                    // This should not happen with the way we parse
+                    // the `session_duration` config value, though.
+                    ctx.config.auth.session_duration.try_into().unwrap_or_else(
+                        |error| match error {
+                            time::error::ConversionRange => panic!("session duration too large"),
+                        }
+                    )
+                ).to_string())
```

Comment:
I think I would prefer `set_cookie` taking a `std::time::Duration` and performing the conversion there. Have `time::Duration` in as few places as possible. 

And also: why the long `unwrap_or_else` instead of a simple `expect("session duration too large")`? In order to notice new error types when updating? Ok but it's highly unlikely that something in that regard will change. So I would just use ```.expect("failed to convert `std::time::Duration` to `time::Duration`")``` or something like that. 

Lastly, isn't it "for the price of NOT being able to represent very long durations"? Also I would just mention `i64` vs `u64` here. Otherwise the comment might make the impression that this precision difference makes any difference on human timescales :D

Reply:
> I think I would prefer set_cookie taking a std::time::Duration and performing the conversion there. Have time::Duration in as few places as possible.

I was thinking the same, *BUT* you were so proud of the encapsulation of the `SessionId` type. We would break that if that did the conversion, because the conversion is only safe because of the way we generate the `Duration`. So if `SessionId` did it, it would make an assumption about where these values come from.

But we can move it if you want, I'm undecided myself.

> In order to notice new error types when updating?

Yes. I mean I can change it if you insist, but I disagree. x)

> Lastly, isn't it "for the price of NOT being able to represent very long durations"?

Not sure. You pay with the ability, not the inability. As in, you exchange the ability for whatever you are getting. Let me reword it to avoid this ambiguity and integrate your other remarks.